tests/unit/test_gce_clients.py

```diff
@@ -0,0 +1,344 @@
+        page2 = copy.deepcopy(crm_one_page_rsp)
+        page2['projects'][0]['projectNumber'] = '3'
+        page2['projects'][1]['projectNumber'] = '4'
+        # This project entry is expected to be filtered out
+        page2['projects'][1]['lifecycleState'] = 'd34d'
+        crm_one_page_rsp['nextPageToken'] = '123token123'
+        with aioresponses() as m:
+            m.get(
+                crm_base_url + 'pageSize=1000',
+                payload=crm_one_page_rsp)
+            m.get(
+                crm_base_url + 'pageSize=1000&pageToken=123token123',
+                payload=page2)
+
+            results = await crm_client.list_all_active_projects()
+
+        expected_rsp = crm_one_page_rsp['projects']
+        expected_rsp.append(page2['projects'].pop(0))
+        assert results == expected_rsp
+
+
+class TestGCPClient:
+    @pytest.fixture
+    def compute_base_url(self):
+        return ('https://www.googleapis.com/compute/v1/projects/'
+                'test-project/aggregated/instances?')
+
+    @pytest.fixture
+    def compute_rsp(self):
+        return {
+            'kind': 'compute#instanceAggregatedList',
+            'id': 'd34dbeef',
+            # Simplified Instance resource
+            'items': {
+                'us-west1-z': {
+                    'instances': [{
+                        'id': '1',
+                        'creationTimestamp': '2018-01-01 00:00:00.0000',
+                        'name': 'instance-1',
+                        'description': 'guc3-instance-1-54kj',
+                        'tags': {
+                            'items': ['some-tag'],
+                            'fingerprint': ''
+                        },
+                        'machineType': 'n1-standard-1',
+                        'status': 'RUNNING',
+                        'statusMessage': 'RUNNING',
+                        'zone': 'us-west9-z',
+                        'canIpForward': False,
+                        'networkInterfaces': [{
+                            'network': 'network/url/string',
+                            'subnetwork': 'subnetwork/url/string',
+                            'networkIP': '192.168.0.1',
+                            'name': 'test-network',
+                            'accessConfigs': [{
+                                'type': 'ONE_TO_ONE_NAT',
+                                'name': 'EXTERNAL NAT',
+                                'natIP': '1.1.1.1',
+                                'kind': 'compute#accessConfig'
+                            }],
+                        }],
+                        'metadata': {
+                            'items': [
+                                {'key': 'default', 'value': 'true'}
+                            ],
+                        },
+                    }],
+                    'warning': {
+                        'warning': 'object'
+                    }
+                }
+            },
+            'nextPageToken':
+            None
```

Comment:
If `'nextPageToken'` key is indeed there even with no pages, then I think you have an extraneous new line here.